src/js/protocols/stm32usbdfu.js

```diff
@@ -775,6 +775,7 @@ STM32DFU_protocol.prototype.upload_procedure = function (step) {
                     for (var j = 0; j < self.flash_layout.sectors[i].num_pages; j++) {
                         if (self.options.erase_chip) {
                             // full chip erase
+                            FC.resetState();
```

Comment:
Calling a high level function from inside a protocol implementation is problematic - especially since this is happening only in a very specific code path. If this is really the correct approach, it will have to be done with a callback.

Reply:
Okey im going to look at this because I think this is the correct approach to can show hide hardwareId when is really needed.